### PR TITLE
feat(Tips): mf-1890 improve tips UI/UX

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-1122
https://mask.atlassian.net/browse/MF-1890
https://mask.atlassian.net/browse/MF-2110

### CODE REVIEW

File: packages/theme/src/Components/Checkbox/index.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import type { FC } from 'react'
+import { Icons } from '@masknet/icons'
+import type { RadioProps } from '../Radio'
+
+export interface CheckboxProps extends RadioProps {}
+
+export const Checkbox: FC<CheckboxProps> = ({ checked, ...rest }) => {
```
Comment (Jack-Works) [inline packages/theme/src/Components/Checkbox/index.tsx]: Please consider using https://react-spectrum.adobe.com/react-aria/useCheckbox.html for better accessibility. e.g. Being able to handle keyboard events.

File: packages/theme/src/Components/Radio/index.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { GeneratedIconProps, Icons } from '@masknet/icons'
+import type { FC } from 'react'
+
+export interface RadioProps extends GeneratedIconProps {
+    checked?: boolean
+}
+
+export const Radio: FC<RadioProps> = ({ checked, ...rest }) => {
```
Comment (Jack-Works) [inline packages/theme/src/Components/Radio/index.tsx]: Please consider using https://react-spectrum.adobe.com/react-aria/useRadioGroup.html for better accessibility. e.g. Being able to handle keyboard events.

File: packages/theme/src/Components/Checkbox/index.tsx (hunk 1 above)
Comment (UncleBill) [reply packages/theme/src/Components/Checkbox/index.tsx]: I think @nuanyang233 could take care of this.

File: packages/theme/src/Components/Radio/index.tsx (hunk 2 above)
Comment (nuanyang233) [reply packages/theme/src/Components/Radio/index.tsx]: The Radio style has been overwritten in the UI kit and can be used.

File: packages/theme/src/Components/Checkbox/index.tsx (hunk 1 above)
Comment (nuanyang233) [reply packages/theme/src/Components/Checkbox/index.tsx]: As with Radio, you can use mui's <Checkbox />

**Review by nuanyang233 — CHANGES_REQUESTED** [review]

File: packages/theme/src/Components/Checkbox/index.tsx (hunk 1 above)
Comment (UncleBill) [reply packages/theme/src/Components/Checkbox/index.tsx]: OK

File: packages/mask/src/components/InjectedComponents/ProfileCard/ProfileCardTitle.tsx [hunk 3]
```diff
@@ -64,12 +64,15 @@ export const ProfileCardTitle: FC<Props> = ({
...
-    const tipAccounts: TipsAccount[] = useMemo(() => {
-        return socialAddressList.map((x) => ({
-            address: x.address,
-            name: x.label,
-            verified: x.type === SocialAddressType.NEXT_ID,
-        }))
+    const tipAccounts = useMemo(() => {
+        return socialAddressList.map(
+            (x): TipsAccount => ({
+                pluginId: NetworkPluginID.PLUGIN_EVM,
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/ProfileCard/ProfileCardTitle.tsx]:
```suggestion
                pluginId: x.networkSupporterPluginID,
```

Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/ProfileCard/ProfileCardTitle.tsx]:
```suggestion
                pluginId: x.networkSupporterPluginID,
```

File: packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx [hunk 4]
```diff
@@ -24,6 +24,19 @@ const useStyles = makeStyles<{}, 'postTipsButton'>()((theme, _, refs) => ({
         },
     },
     postTipsButton: {},
+    roundButton: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        borderWidth: 1,
+        borderStyle: 'solid',
+        borderColor: theme.palette.mode === 'dark' ? '#536471' : '#d2dbe0',
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx]: No dead-code color.

File: packages/mask/src/plugins/Tips/components/NFTSection/NFTList.tsx [hunk 5]
```diff
@@ -98,19 +99,23 @@ interface NFTItemProps {
...
+                url={token.metadata?.imageURL ?? token.metadata?.mediaURL}
+                tokenId={token.tokenId}
+                classes={{
+                    fallbackImage: classes.fallbackImage,
+                    iframe: classes.assetPlayerIframe,
+                    wrapper: classes.wrapper,
+                    imgWrapper: classes.imgWrapper,
+                }}
+            />
+            <div>#{Others?.formatAddress(token.tokenId)}</div>
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/components/NFTSection/NFTList.tsx]: ?

File: packages/web3-shared/base/src/specs/index.ts [hunk 6]
```diff
@@ -724,6 +724,8 @@ export interface TransactionChecker<ChainId, Transaction> {
 }
 
 export interface ConnectionOptions<ChainId, ProviderType, Transaction> {
+    /** Designate the plugin id of the transaction. */
+    pluginId?: NetworkPluginID
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts]: Not used?

File: packages/shared/src/hooks/useImageBase64.ts [hunk 7]
```diff
@@ -22,6 +22,7 @@ const responseToBase64 = async (response: Response) => {
 }
 
 export function useImageBase64(key = '', url?: string) {
+    key = key || (url ?? '')
```
Comment (guanbinrui) [inline packages/shared/src/hooks/useImageBase64.ts]: Don't touch the parameters.

File: packages/mask/src/plugins/Tips/components/TipTaskManager.tsx [hunk 8]
```diff
@@ -34,14 +36,26 @@ export const TipTaskManager: FC<React.PropsWithChildren<{}>> = ({ children }) =>
...
     return (
-        <>
-            {tasks.map((task) => (
-                <TipTaskProvider key={task.id} task={task}>
-                    <TipDialog open key={task.id} onClose={() => removeTask(task)} />
-                </TipTaskProvider>
-            ))}
+        <PluginWeb3ContextProvider pluginID={pluginId} value={{ chainId: targetChainId }}>
+            {tasks.map((task) => {
+                const tipsAccount = task.addresses.find((x) => x.address === task.recipient)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/components/TipTaskManager.tsx]: isSameAddress

File: packages/mask/src/plugins/Tips/components/common/ConfirmModal.tsx [hunk 9]
```diff
@@ -51,8 +97,56 @@ export const ConfirmModal: FC<Props> = ({ className, message, icon, confirmText,
...
+                                        renderOrder={0}
+                                        pluginID={pluginId}
+                                    />
+                                </div>
+                                <div className={classes.nftName}>
+                                    <Typography fontWeight={700} fontSize={20} lineHeight="24px">
+                                        {nonFungibleToken?.metadata?.name}
+                                    </Typography>
+                                    <Typography fontWeight={700} fontSize={16} mx="7px">
+                                        #{nonFungibleToken?.tokenId}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/components/common/ConfirmModal.tsx]: Others.formatTokenId

File: packages/mask/src/plugins/Tips/components/common/ConfirmModal.tsx [hunk 10]
```diff
@@ -51,8 +97,56 @@ export const ConfirmModal: FC<Props> = ({ className, message, icon, confirmText,
...
+                                    </Typography>
+                                    <Icons.Verified size={21.43} />
+                                </div>
+                            </>
+                        ) : null}
+                        <Typography className={classes.congratulation} mt="24px">
+                            Congratulations!
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/components/common/ConfirmModal.tsx]: i18n

File: packages/mask/src/plugins/Tips/components/NFTSection/NFTList.tsx (hunk 5 above)
Comment (UncleBill) [reply packages/mask/src/plugins/Tips/components/NFTSection/NFTList.tsx]: Removed.

File: packages/shared/src/hooks/useImageBase64.ts (hunk 7 above)
Comment (UncleBill) [reply packages/shared/src/hooks/useImageBase64.ts]: Not modifying an object parameter.

If we don't want this, I would setup a eslint rule to restrict.

File: packages/web3-shared/base/src/specs/index.ts (hunk 6 above)
Comment (UncleBill) [reply packages/web3-shared/base/src/specs/index.ts]: ~~Indeed. Removed.~~

```ts
        const web3 = await connection.getWeb3({
            pluginId: NetworkPluginID.PLUGIN_EVM, // Here.
            chainId,
            account,
        })
```

File: packages/shared/src/hooks/useImageBase64.ts (hunk 7 above)
Comment (UncleBill) [reply packages/shared/src/hooks/useImageBase64.ts]: 🤷‍♂️ 
image

File: packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx (hunk 4 above)
Comment (UncleBill) [reply packages/mask/src/plugins/Tips/SNSAdaptor/components/TipsRealmContent/index.tsx]: To match Twitter's color.

File: packages/plugin-infra/src/web3/useAddressType.ts [hunk 11]
```diff
@@ -13,5 +13,5 @@ export function useAddressType<S extends 'all' | void = void, T extends NetworkP
     return useAsyncRetry<Web3Helper.AddressTypeScope<S, T> | undefined>(async () => {
         if (!address) return
         return connection?.getAddressType?.(address, options)
-    }, [address])
+    }, [address, options?.chainId])
```
Comment (guanbinrui) [inline packages/plugin-infra/src/web3/useAddressType.ts]:
```suggestion
    }, [address, connection, JSON.stringify(options)])
```

File: packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/index.tsx [hunk 12]
```diff
@@ -101,22 +101,26 @@ export function SelectProviderDialog(props: SelectProviderDialogProps) {
     )
 
     // not available for the native app
+
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Wallet/SNSAdaptor/SelectProviderDialog/index.tsx]:
```suggestion
```

File: packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx [hunk 13]
```diff
@@ -30,35 +36,96 @@ const useStyles = makeStyles()((theme) => ({
...
+                    provider={provider}
+                    readonly={readonly}
+                    renderOrder={renderOrder}
+                    pluginID={pluginID}
+                />
+                <div className={classes.description}>
+                    <Typography className={classes.name} color="textPrimary" variant="body2">
+                        {uiTokenId}
+                    </Typography>
+                </div>
```
Comment (guanbinrui) [inline packages/mask/src/extension/options-page/DashboardComponents/CollectibleList/CollectibleItem.tsx]:
```suggestion
        const uiTokenId = Others?.formatTokenId(asset.tokenId, 4)

        const SelectableButton = selectable && multiple ? Checkbox : Radio

        return (
            <div
                className={cx(classes.card, className, {
                    [classes.inactive]: inactive,
                })}
                {...rest}
                ref={ref}>
                <CollectibleCard
                    className={classes.collectibleCard}
                    asset={asset}
                    provider={provider}
                    readonly={readonly}
                    renderOrder={renderOrder}
                    pluginID={pluginID}
                />
                {
                    uiTokenId ? (
                        <div className={classes.description}>
                            <Typography className={classes.name} color="textPrimary" variant="body2">
                                {uiTokenId}
                            </Typography>
                        </div>
                    ) : null
                }
```

File: packages/mask/src/plugins/Tips/components/TipTaskManager.tsx [hunk 14]
```diff
@@ -34,14 +37,28 @@ export const TipTaskManager: FC<React.PropsWithChildren<{}>> = ({ children }) =>
...
-        <>
-            {tasks.map((task) => (
-                <TipTaskProvider key={task.id} task={task}>
-                    <TipDialog open key={task.id} onClose={() => removeTask(task)} />
-                </TipTaskProvider>
-            ))}
+        <PluginWeb3ContextProvider pluginID={pluginId} value={{ chainId: targetChainId }}>
+            {tasks.map((task) => {
+                const tipsAccount = task.addresses.find(
+                    (x) => Others?.isSameAddress(x.address, task.recipient) ?? x.address === task.recipient,
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/components/TipTaskManager.tsx]:
```suggestion
                    (x) => Others?.isSameAddress(x.address, task.recipient),
```

**Review by guanbinrui — DISMISSED** [review]

File: packages/mask/src/plugins/Tips/components/RecipientSelect.tsx [hunk 15]
```diff
@@ -1,65 +1,200 @@
...
+            filter: 'drop-shadow(0px 4px 10px rgba(28, 104, 243, 0.2))',
+        },
+        tooltip: {
+            maxWidth: 'unset',
         },
     }
 })
 
-export const RecipientSelect: FC = () => {
+const pluginIconMap: Record<NetworkPluginID, ReactNode> = {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/components/RecipientSelect.tsx:66]: This could be a shared component.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**